### PR TITLE
`UnitControl`: show unit label when `units` prop has only one unit

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Fix an issue in `UnitControl`, where the unit label was not being shown if the `units` array prop only had one item.

## Why?

This is a bug and should be fixed. See https://github.com/WordPress/gutenberg/pull/40697#discussion_r861178483 and https://github.com/WordPress/gutenberg/pull/40697#issuecomment-1115524503 for more context.

## How?

The fix is to use the label of the single unit passed through the `units` prop as the initial unit value (as suggested in this [comment](https://github.com/WordPress/gutenberg/pull/40697#issuecomment-1115524503) by @stokesman ).

This PR also restores the `should render label if single units` unit test to how it used to be before the changes applied in #40697.

## Testing Instructions

- Make sure tests pass

- Visit the[ "Single Unit Controlled"](http://localhost:50240/?path=/story/components-experimental-unitcontrol--with-single-unit-controlled) Storybook example on `trunk`, verify that there's no unit being displayed when the Story initially renders. Check that the unit is only shown after a numeric value is entered in the input field.

- Apply this patch below to restore the same Storybook example on this branch, visit the same "Single Unit Controlled" Storybook example, and verify that the initial unit is being displayed during the Story initial render.

<details>

<summary>Click to show the patch</summary>

```diff
diff --git a/packages/components/src/unit-control/stories/index.tsx b/packages/components/src/unit-control/stories/index.tsx
index 87dfd3a84c..de60479e7b 100644
--- a/packages/components/src/unit-control/stories/index.tsx
+++ b/packages/components/src/unit-control/stories/index.tsx
@@ -117,6 +117,29 @@ WithSingleUnit.args = {
 	units: CSS_UNITS.slice( 0, 1 ),
 };
 
+export const WithSingleUnitControlled: ComponentStory<
+	typeof UnitControl
+> = ( { onChange, ...args } ) => {
+	const [ value, setValue ] = useState< string | undefined >( undefined );
+
+	return (
+		<div style={ { maxWidth: '100px' } }>
+			<UnitControl
+				{ ...args }
+				value={ value }
+				onChange={ ( v, extra ) => {
+					setValue( v );
+					onChange?.( v, extra );
+				} }
+			/>
+		</div>
+	);
+};
+WithSingleUnitControlled.args = {
+	...Default.args,
+	units: CSS_UNITS.slice( 0, 1 ),
+};
+
 /**
  * It is possible to pass a custom list of units. Every time the unit changes,
  * if the `isResetValueOnUnitChange` is set to `true`, the input's quantity is
```
</details>

## Screenshots or screencast 

**Before**

https://user-images.githubusercontent.com/1083581/166507458-919ec3c5-022f-4570-b7ff-5f9807f598a6.mp4

**After**

https://user-images.githubusercontent.com/1083581/166507590-03fa417d-1585-4fc8-b84f-81c4efda539e.mp4

### CODE REVIEW

File: packages/components/src/unit-control/stories/index.tsx [hunk 1]
```diff
@@ -109,40 +109,10 @@ TweakingTheNumberInput.args = {
 /**
  * When only one unit is available, the unit selection dropdown becomes static text.
  */
-export const WithSingleUnitControlled: ComponentStory<
```
Comment (ciampo) [inline packages/components/src/unit-control/stories/index.tsx:112]: Just undoing some Storybook changes that were left from previous work on #40697

File: packages/components/src/unit-control/test/index.tsx [hunk 2]
```diff
@@ -135,12 +135,7 @@ describe( 'UnitControl', () => {
 		} );
 
 		it( 'should render label if single units', () => {
-			render(
-				<UnitControl
-					units={ [ { value: '%', label: '%' } ] }
-					value="30%"
```
Comment (ciampo) [inline packages/components/src/unit-control/test/index.tsx:141]: Removing `value` prop from test, which was added as a temporary fix in #40697 (see https://github.com/WordPress/gutenberg/pull/40697#discussion_r861178483 for context)

File: packages/components/src/unit-control/index.tsx [hunk 3]
```diff
@@ -88,7 +88,7 @@ function UnforwardedUnitControl(
 	);
 
 	const [ unit, setUnit ] = useControlledState< string | undefined >(
-		unitProp,
+		units.length === 1 ? units[ 0 ].label : unitProp,
```
Comment (aaronrobertshaw) [inline packages/components/src/unit-control/index.tsx]:
```suggestion
		units.length === 1 ? units[ 0 ].value : unitProp,
```

Should this not reference the unit's `value` as `label` could be anything? Not sure it differs in most cases for the web, native though uses human readable strings for the labels.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for cleaning this up @ciampo 👍 

✅ Tests pass
✅ All Storybook examples for `UnitControl` function as expected
✅ Works as expected in editor and theme setting only single unit

I did have one minor question/tweak that I've left as an inline comment. Happy to review again and give final approval once that's resolved.

File: packages/components/src/unit-control/index.tsx (hunk 3 above)
Comment (ciampo) [reply packages/components/src/unit-control/index.tsx]: Makes sense to me, good catch!

**Review by aaronrobertshaw — APPROVED** [review]
LGTM 🚢 

✅ Tests still pass
✅ UnitControl behaves as advertised within Storybook examples
✅ Still working in the editor with varying custom unit configurations

I'd say this should be right to land once all the checks are green. 👍

**Pull request merged**